### PR TITLE
Add unit tests for operator

### PR DESCRIPTION
**Pull request by anishasthana**
Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

### CODE REVIEW

File: controllers/mcad_controller_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	mcadCRCase1    = "./testdata/mcad_test_cases/case_1.yaml"
+	mcadConfigMap1 = "./testdata/mcad_test_results/case_1/configmap.yaml"
+)
+
+func deployMCAD(ctx context.Context, path string, opts mf.Option) {
+	mcad := &codeflarev1alpha1.MCAD{}
+	err := convertToStructuredResource(path, mcad, opts)
+	Expect(err).NotTo(HaveOccurred())
+	Expect(k8sClient.Create(ctx, mcad)).Should(Succeed())
+}
```
Comment (KPostOffice) [inline controllers/mcad_controller_test.go:26]: Does OpenShift CI deploy a cluster when running tests? This is definitely more of an integration test unless I'm missing a mock somewhere.

Comment (anishasthana) [reply controllers/mcad_controller_test.go:26]: This is actually mocking the api, not deploying real objects on an actual cluster.

Comment (anishasthana) [reply controllers/mcad_controller_test.go:26]: You can try running the test yourself via `make test-unit` on an environment that isn't logged in on any OpenShift/k8s cluster.

**Comment by anishasthana** [comment]
Closes #1 and #3

File: controllers/testutil/utils.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+	v1 "k8s.io/api/core/v1"
+	"reflect"
+)
+
+func notEqualMsg(value string) {
+	print(fmt.Sprintf("%s are not equal.", value))
+}
+
+func ConfigMapsAreEqual(expected v1.ConfigMap, actual v1.ConfigMap) bool {
+	if expected.Name != actual.Name {
```
Comment (KPostOffice) [inline controllers/testutil/utils.go:14]: Why can't we just do `DeepEqual` on `expected`?

Comment (KPostOffice) [reply controllers/testutil/utils.go:14]: Test fails without it, I don't really understand why

**Review by KPostOffice — APPROVED** [review]
/lgtm

**Pull request merged**